apps/web-giddh/src/app/new-inventory/component/bulk-stock-edit/bulk-stock-edit.component.ts

```diff
@@ -0,0 +1,918 @@
+            skuCode: [controlValue.skuCode, Validators.required],
+            archive: [controlValue.archive, Validators.required],
+            taxes: [controlValue.taxes, Validators.required]
+        })
+    }
+
+    /**
+     * The addRow(data) is used to push addNewRow() in "bulkStockEditForm" Formgroup
+     * @private
+     * @param data
+     */
+    private addRow(data) {
+        this.bulkStockData.push(this.addNewRow(data));
+    }
+    /**
+      * Getter method for "fields" of "bulkStockEditForm" Formgroup
+      */
+    get bulkStockData(): FormArray {
+        return this.bulkStockEditForm.get("fields") as FormArray
+    }
+    /**
+     * NgFormSubmit for "bulkStockEditForm" Formgroup
+     */
+    onFormSubmit() {
+        // console.log(this.bulkStockEditForm.value);
+    }
+
+    /**
+    * Get select value from select field
+    */
+    public setPaginationData(data: any): void {
+        this.pagination = {
+            currentPage: data?.page,
+            itemsPerPage: data?.count,
+            totalPages: data?.totalPages,
+            totalItems: data?.totalItems
+        }
+    }
+
+    /**
+     * Get select value from filter checkbox
+     * @param Event
+     */
+    public showSelectedHeaderColumns(e): void {
+        // console.log("Event: ", e);
+    }
+    /**
+  * Get Pagination page change event
+  */
+    public pageChanged(e): void {
+        if (this.pagination.currentPage !== e?.page) {
+            this.isLoading = true;
+            this.store.dispatch(this.inventoryAction.GetBulkStockList({
+                inventoryType: this.inventoryType, page: e.page, count: this.pageCount, body: {
+                    "search": this.searchString !== null ? this.searchString : "",
+                    "searchBy": this.searchStringKey !== null ? this.searchStringKey : "",
+                    "sortBy": this.advanceSearchData !== null ? this.advanceSearchData?.sortBy : (this.sortOrderKey !== null ? this.sortOrderKey : ""),
+                    "sort": this.sortOrderStatus !== null ? this.sortOrderStatus : "asc",
+                    "expression": this.advanceSearchData !== null ? this.advanceSearchData?.expression : "",
+                    "rate": this.advanceSearchData !== null ? this.advanceSearchData?.amount : ""
+                }
+            }));
+        }
+    }
+
+    /**
+     * toggleInput(key) is used to change boolean value of tableHeadInput 
+     * object key value to show hide table head input fields
+     * @param key
+     */
+    public toggleInput(key): void {
+        this.hideTableHeadInput();
+        this.tableHeadInput[key] = !this.tableHeadInput[key];
+    }
+
+    /**
+     * This will false all the keys of tableHeadInput Object 
+     * which will hide all the input fields from table head on, this method trigger on click outside the table head
+     */
+    public hideTableHeadInput(): void {
+        Object.entries(this.tableHeadInput).forEach(([key]) => {
```

Comment:
why is key taken as array?

Reply:
Its mis typed, there is no login behind this i will correct this also